examples/coap-client.c

```diff
@@ -1797,11 +1811,25 @@ main(int argc, char **argv) {
 #endif
 
   while(!quit &&
-        !(ready && !tracked_tokens_count && !is_mcast && coap_can_exit(ctx))) {
+        !(ready && !tracked_tokens_count && !is_mcast && !repeat_count &&
+          coap_can_exit(ctx)) ) {
+    uint32_t timeout_ms;
+    /*
+     * 3 factors determine how long to wait in coap_io_process()
+     *   Remaining overall wait time (wait_ms)
+     *   Remaining overall observe unsolicited response time (obs_ms)
+     *   Delay of up to one second before sending off the next request
+     */
+    if (obs_ms) {
+      timeout_ms = min(wait_ms, obs_ms);
+    } else {
+      timeout_ms = wait_ms;
+    }
+    if (repeat_count) {
+      timeout_ms = min((int)timeout_ms, repeat_ms);
```

Comment:
At this point, `repeat_ms` should not be negative. Wouldn't it make more sense to cast `repeat_ms` to `uint32_t`? Possibly with `assert(repeat_ms >= 0)`? (See also comment addressing the datatype.)

Reply:
`repeat_ms `is now `uint32_t`.  `assert(repeat_ms >= 0)` will always be true.